defopt.py

```diff
@@ -118,8 +118,23 @@ def _create_parser(funcs, *args, **kwargs):
     return parser
 
 
-class _Formatter(RawTextHelpFormatter, ArgumentDefaultsHelpFormatter):
-    pass
+class _Formatter(RawTextHelpFormatter):
+
+    # modified from ArgumentDefaultsHelpFormatter to add type information, and
+    # remove defaults for varargs (`ZERO_OR_MORE`).
+    def _get_help_string(self, action):
+        info = []
+        if action.type is not None and '%(type)' not in action.help:
+            info.append('%(type)s')
+        if (action.default != argparse.SUPPRESS
+                and '%(default)' not in action.help
+                and action.option_strings
+                and action.nargs != argparse.ZERO_OR_MORE):
```

Comment:
I think checking `if action.default` is a step backwards, since now it's hiding any falsy value, including `[]`, `None` and `False`.

If we're only trying to hide the varargs case, perhaps we can avoid having to mess with this too much by actually using `argparse.SUPPRESS` as the default for those, and setting a parser-level default with `parser.set_defaults` (or just tolerate the missing attribute in `_call_function`).